### DISCUSSION TITLE
Sulu still shows a "-1" in page urls

### DISCUSSION THREAD

**Discussion by nlipkanov** [sulu/sulu#6891] [Q&A] [answered]
| Q | A |
| --- | --- |
| Sulu Version | ^5.4
| PHP Version | ^8.0
| DB Version | MARIADB-10.7.4
| Browser Version | Google Chrome 105.0.5195.127
# Actual Behavior

Duplicated pages have a "-1" string attatched to their routes end, e.g. about-1.
After deleting one of the duplicated page, the string "-1" is still present.

# Expected Behavior

Renaming the duplicated pages resource locator / route should remove the attatched string "-1".

# Steps to Reproduce

1. Create a page in sulu named "test".
2. Duplicate that page and name it something else, e.g. "test new".
3. Then take a look at the resource locators of both pages. One of them must have "-1" string attatched to it.

**Comment by alexander-schranz** [comment]
@nlipkanov renaming a page does do nothing with the URL. If you want to generate a new url after you changed the titleyou need to click the `Refresh URL` button:

![refresh-url](https://user-images.githubusercontent.com/1698337/195794663-9f854266-9dfd-4b9e-b223-d2674461be79.gif)

So this behaviour is expected and not a bug.

**Comment by nlipkanov** [comment]
In my case, the "-1" still stays and does not go away if you visit the page in the frontend. In fact, when I rename the title to something else multiple times, then it shows a "-2" in the url of the page.

**Comment by alexander-schranz** [comment, marked as answer]
@nlipkanov are you under `dev` or `prod`. Are you using the correct `phpcr` property. Old versions had a `path` which is incorrect, as that is the PHPCR path which has nothing todo with the `Url`. Thats also new versions not longer have that property so check your twig templates if they have correctly used the `.url` property.

**Comment by nlipkanov** [comment]
I am under dev.

**Comment by nlipkanov** [comment]
I also get an error when I try to rename the resource locator:
`Resource Locator was moved to \"/leistungen-2\" (2f3ee38f-36a1-426c-aede-dfd6f795cc50)`

I do not have any page which has a resource locator like "leistungen" or "leistungen-1". I assume this is some caching issue.
Weird enough that I do not have this issue on my local machine.

**Comment by alexander-schranz** [comment]
@nlipkanov it can still be that a page has that url in the `Url History`. You can use the `bin/console doctrine:phpcr:shell` and analyse your url tree under `/cmf/<webspace_key>/routes`.

If you think its a cache issue maybe clear all your cache pools. But normally no caches are active under `dev`. And mostly the errors lay in false outputted url in the twig template.

If you are under `prod` check the cache configs of your application `sulu_document_manager.yaml`. Also make sure you use the latest version of it 2.4.5 or 2.5.1. If you are on older version please update your packages. Don't forget to also update the `config/packages/` files maybe something wrong there in your project.

**Comment by nlipkanov** [comment]
So, apparently it was just the url history that was in my way. Thanks for pointing that out!
I deleted the urls in the history and everything works as expected!
